LittleToySourceGenerator/LittleToySourceGenerator.cs

```diff
@@ -587,352 +622,385 @@ private static string GetDotsnetTypeName(ITypeSymbol typeSymbol)
 
-            if (hasRemovedEventViewTypes)
+            return false;
+        });
+
+        if (ifAdd)
+        {
+            GenerateEventSystemOnAddedEvents(addedEventViewTypes, eventComponentType, classModel, onCreateMethodModel, onUpdateMethodModel);
+            GenerateEventSystemOnAddedEventJobs(eventComponentType, classModel);
+        }
+
+        var ifRemove = classCandidates.Any(viewType =>
+        {
+            var viewTypeSymbol = (ITypeSymbol)context.Compilation.GetSemanticModel(viewType.SyntaxTree).GetDeclaredSymbol(viewType);
+            foreach (var a in viewTypeSymbol.GetCustomAttributes(true))
             {
-                builder.AppendLine($"_removedComponentArrayDataROComponentTypeHandle.Update(this);");
-                foreach (var eventViewType in removedEventViewTypes)
+                if (a.IsAttribute(OnRemovedEventViewAttributeType))
                 {
-                    builder.AppendLine();
-                    builder.AppendLine($"var notify{eventViewType.Name}Remove{typeSymbol.Name}Job = new NotifyRemove{typeSymbol.Name}Job<{eventViewType.Name}>");
-                    builder.OpenBraces();
-                    builder.AppendLine($"EntityManager = this.EntityManager,");
-                    builder.AppendLine($"DataTypeHandle = _removedComponentArrayDataROComponentTypeHandle,");
-                    builder.AppendLine($"ListenerTypeHandle = EntityManager.GetComponentTypeHandle<{eventViewType.Name}>(false),");
-                    builder.CloseBraces(";");
-                    builder.AppendLine($"CompleteDependency();");
-                    builder.AppendLine($"JobEntityBatchExtensions.RunWithoutJobs(ref notify{eventViewType.Name}Remove{typeSymbol.Name}Job, _entityWith{eventViewType.Name}AndRemovedComponentArrayDataQuery);");
+                    if (a.GetFieldValueTypes("Types").Contains(eventComponentType))
+                    {
+                        return true;
+                    }
                 }
             }
 
-            builder.CloseBraces();
-            if (hasDirtyEventViewTypes)
-            {
-                builder.AppendLine();
-                GenerateNotifyDirtyJob();
-                builder.AppendLine();
-                GenerateResetDirtyJob();
-            }
+            return false;
+        });
 
-            if (hasAddedEventViewTypes)
-            {
-                builder.AppendLine();
-                GenerateNotifyAddJob();
-            }
+        if (ifRemove)
+        {
+            GenerateEventSystemOnRemovedEvents(removedEventViewTypes, eventComponentType, classModel, onCreateMethodModel, onUpdateMethodModel);
+            GenerateEventSystemOnRemovedEventJobs(eventComponentType, classModel);
+        }
 
-            if (hasRemovedEventViewTypes)
-            {
-                builder.AppendLine();
-                GenerateNotifyRemoveJob();
-            }
+        if (!ifDirty && !ifAdd && !ifRemove)
+        {
+            //return null;
+        }
```

Comment:
what is the purpose of those three lines pls?

Reply:
Ooh, that one. I should uncomment it. Thanks for spotting it.